pywbem_mock/_baserepository.py

```diff
@@ -0,0 +1,353 @@
+
+        Returns:
+            Python iterator for the object values in the object store.
+            If copy == True, each object is copied before it is returned.
+        """
+        pass
+
+    @abstractmethod
+    def len(self):
+        """
+        Get count of objects in this object store.
+
+        Returns:
+            Integer that is count of objects in this objectstore
+
+        """
+        pass
+
+
+@six.add_metaclass(ABCMeta)
+class BaseRepositoryAPI(object):
+    """
+    An abstract base class defining the required  APIs to provide access to a
+    data store for a CIM repository.  The API provides functions to:
+
+    1. Manage CIM namespaces in the data repository including creation, deletion
+       and getting a list of the available namespaces.
+    2. Accessing the object store within the repository for the objects of the
+       following CIM types: (CIM classes, CIM instances, and CIM qualifier
+       decelarations) so that methods of the ObjStoreAPI can be used to access
+       objects by namespace so that CIM objects can be manupulated in the
+       repository by namespace.
+    """
+
+    @compatibleabstractproperty
+    def namespaces(self):
+        """
+        Readonly property that returns a list with the names of the
+        namespaces defined for this repository
+
+        Returns:
+            list of :term:`string` items containing the namespace names
+        """
+        pass
+
+    @abstractmethod
+    def validate_namespace(self, namespace):
+        """
+        Validate if the namespace parameter exists in the repository. If the
+        namespace does not exist a KeyError is raised.
+
+        Raises:
+            KeyError: If the namespace is not defined in the repository
+        """
+        pass
+
+    @abstractmethod
+    def add_namespace(self, namespace):
+        """
+        Add a CIM namespace to the repository.
+
+        The namespace must not yet exist in the mock repository.
+
+        The default connection namespace is automatically added to
+        the mock repository upon creation of this connection.
+
+          Parameters:
+            namespace (:term:`string`):
+              The name of the CIM namespace in the mock repository. Must not be
+              `None`. Leading and trailing slash characters are split off
+              from the provided string.
+
+          Raises:
+            ValueError: If the namespace argument is None or the
+             namespace already exists.
+        """
+        pass
+
+    @abstractmethod
+    def remove_namespace(self, namespace):
+        """
+        Remove a CIM namespace from the repository.
+
+        The namespace must exist in the repository and must be empty.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            The name of the CIM namespace in the mock repository. Must not be
+            `None`. Any leading and trailing slash characters are split off
+            from the provided string.
```

Comment:
For a method that takes a namespace name as input to look it up, we would not say leading and trailing slashes are split off, but instead that they are ignored when looking up the namespace.
Also, we should say that namespaces are looked up case insensitively.

Reply:
DONE for add, remove